tests/codegen/asm/critical.rs

```diff
@@ -0,0 +1,27 @@
+//@ only-x86_64
+//@ compile-flags: -C no-prepopulate-passes
+#![feature(asm_goto)]
+#![feature(asm_goto_with_outputs)]
+#![crate_type = "lib"]
+use std::arch::asm;
+
+// Regression test for #137867. Check that critical edges have been split before code generation,
+// and so all labels in callbr lead to distinct blocks (CHECK-DAG matches are non-overlapping).
+//
+// CHECK-LABEL: @f(
+// CHECK:       callbr i32 asm
+// CHECK-NEXT:  to label %[[BB0:.*]] [label %[[BB1:.*]], label %[[BB2:.*]]],
+// CHECK-DAG:   [[BB0]]:
+// CHECK-DAG:   [[BB1]]:
+// CHECK-DAG:   [[BB2]]:
```

Comment:
Can you assert that the jump instructions are correct? Even before this PR it would already result in branching to different blocks. It was just that one of the target blocks then jumps to another of the target blocks rather than both target blocks jumping to a common block.

Reply:
That only has the same target twice because of optimizations. With `-Zmir-opt-level=0` I get two different targets on nightly: https://rust.godbolt.org/z/xvasqP79e